src/components/SearchBar/SearchBar.component.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import styled from 'styled-components';
+import { useSearch } from '../../providers/SearchContext';
+
+const SearchbarContainer = styled.div`
+  padding: 0 2rem;
+`;
+
+const Searchbar = styled.div`
+  display: flex;
+  align-items: center;
+  background-color: seashell;
+  border-radius: 5px;
+`;
+
+const InputSearchbar = styled.input`
+  width: 11rem;
+  border: none;
+  &:focus {
+    outline: none;
+  }
+`;
+
+const SearchBar = () => {
+  const { query, setQuery } = useSearch();
+
+  return (
+    <SearchbarContainer>
+      <Searchbar>
+        <span className="material-icons-outlined">search</span>
+        <InputSearchbar
+          type="text"
+          placeholder="Pokemon"
+          value={query}
+          onChange={(e) => setQuery(e.target.value)}
```

Comment:
This component shouldn't be the one calling `setQuery`, this should be in a parent component and this component should call a handler without knowing what the handler does